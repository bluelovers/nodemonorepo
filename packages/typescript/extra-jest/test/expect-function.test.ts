import expectFunction from '../lib/expect-function'

describe('when matched function throws an error', () => {
  const error = 'error'
  const fn = () => { throw error }

  it('matches against thrown error', () => {
    expectFunction(fn).throws.toBe(error)
  })

  it('matches thrown error against snapshot', () => {
    expectFunction(fn).throws.toMatchSnapshot()
  })

  it('throws an error for trying to get a return value', () => {
    expectFunction(() => expectFunction(fn).returns).throws.toMatchSnapshot()
  })
})

describe('when matched function does not throw an error', () => {
  const value = 'value'
  const fn = () => value

  it('matches against returning value', () => {
    expectFunction(fn).returns.toBe(value)
  })

  it('matches returning value against snapshot', () => {
    expectFunction(fn).returns.toMatchSnapshot()
  })

  it('throws an error for trying to get an error', () => {
    expectFunction(() => expectFunction(fn).throws).throws.toMatchSnapshot()
  })
})
