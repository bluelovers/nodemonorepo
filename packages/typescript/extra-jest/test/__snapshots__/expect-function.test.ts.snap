// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when matched function does not throw an error matches returning value against snapshot 1`] = `"value"`;

exports[`when matched function does not throw an error throws an error for trying to get an error 1`] = `[Error: Expecting the matched function to throw an error, but it returns a value: value]`;

exports[`when matched function throws an error matches thrown error against snapshot 1`] = `"error"`;

exports[`when matched function throws an error throws an error for trying to get a return value 1`] = `[Error: Expecting the matched function to return a value, but it throws an error: error]`;
